test/workflow-orchestration.ts

```diff
@@ -660,4 +658,103 @@ describe('When a request spans multiple CMR pages', function () {
       });
     });
   });
+
+  describe('and the number of granules returned by the CMR changes while paging', function () {
```

Comment:
I'm more interested in seeing that the request queues the right things and eventually completes the request.

I'd change to a small number of granules like initial 5 and then updated to 2 and change the page size to something small and then simulate through work items completing.

Also add a test going in the other direction starting with 2 and then finishing with 3.

I'm thinking the case of a synchronous request with one granule getting another granule is unlikely so if we blow up when that happens it's ok.

Reply:
Note from my other comment that I think we still need the test of 2 going to 3, but the request should only process the first 2 granules and complete rather than trying to process the 3rd granule.

Reply:
Done